### PR TITLE
lint: add support for other linters

### PR DESCRIPTION
**Pull request by moshe010**

### CODE REVIEW

File: pkg/daemon/daemon.go [hunk 1]
```diff
@@ -2,18 +2,19 @@ package daemon
 
 import (
 	"encoding/json"
-	"k8s.io/apimachinery/pkg/types"
 	"net"
 	"strings"
 	"time"
 
+	"k8s.io/apimachinery/pkg/types"
```
Comment (Mmduh-483) [inline pkg/daemon/daemon.go]: It should be in the 3rd party section (the last one).

File: pkg/daemon/daemon.go [hunk 2]
```diff
@@ -2,18 +2,19 @@ package daemon
...
+
 	"github.com/Mellanox/ib-kubernetes/pkg/guid"
 	k8sClient "github.com/Mellanox/ib-kubernetes/pkg/k8s-client"
 	"github.com/Mellanox/ib-kubernetes/pkg/sm"
 	"github.com/Mellanox/ib-kubernetes/pkg/sm/plugins"
 	"github.com/Mellanox/ib-kubernetes/pkg/utils"
 	"github.com/Mellanox/ib-kubernetes/pkg/watcher"
-	resEvenHandler "github.com/Mellanox/ib-kubernetes/pkg/watcher/resouce-event-handler"
+	resEvenHandler "github.com/Mellanox/ib-kubernetes/pkg/watcher/resource-event-handler"
```
Comment (Mmduh-483) [inline pkg/daemon/daemon.go]: IMHO fixing the name of the `resource-event-handler` pkg should be in a separate commit.

File: Makefile [hunk 3]
```diff
@@ -66,12 +65,6 @@ $(BUILDDIR)/$(BINARY_NAME): $(GOFILES) | $(BUILDDIR)
 
 # Tools
 
-.PHONY: fmt
-fmt: ; $(info  running gofmt...) @ ## Run gofmt on all source files
-	@ret=0 && for d in $$($(GO) list -f '{{.Dir}}' ./... | grep -v /vendor/); do \
-		$(GOFMT) -l -w $$d/*.go || ret=$$? ; \
-	 done ; exit $$ret
-
```
Comment (Mmduh-483) [inline Makefile]: why remove fmt?

File: cmd/ib-kubernetes/main.go [hunk 4]
```diff
@@ -4,8 +4,9 @@ import (
 	"flag"
 	"os"
 
-	"github.com/Mellanox/ib-kubernetes/pkg/daemon"
 	"github.com/golang/glog"
+
+	"github.com/Mellanox/ib-kubernetes/pkg/daemon"
```
Comment (Mmduh-483) [inline cmd/ib-kubernetes/main.go]: line 7 `"github.com/Mellanox/ib-kubernetes/pkg/daemon"` and line 9 `"github.com/Mellanox/ib-kubernetes/pkg/daemon"` should be swapped, `glog` is 3rd party pkg

File: pkg/daemon/daemon.go (hunk 1 above)
Comment (Mmduh-483) [inline pkg/daemon/daemon.go]: Should be moved to the 3rd party section (the last one).

File: pkg/drivers/http/client.go [hunk 5]
```diff
@@ -76,7 +76,7 @@ func (c *client) createRequest(method, url string, body io.Reader) (*http.Reques
 		}
 		req.SetBasicAuth(c.basicAuth.Username, c.basicAuth.Password)
 	} else {
-		err = fmt.Errorf("createRequest(): unkown atuhentication mode %v", c.authMode)
+		err = fmt.Errorf("createRequest(): unknown authentication mode %v", c.authMode)
```
Comment (Mmduh-483) [inline pkg/drivers/http/client.go]: this is not related to the linting, it can be moved to other commit

File: pkg/drivers/http/client.go [hunk 6]
```diff
@@ -94,6 +94,7 @@ func (c *client) executeRequest(method, url string, expectedStatusCode int, body
 	if err != nil {
 		return nil, fmt.Errorf("faied request %v", err)
 	}
+	defer resp.Body.Close()
```
Comment (Mmduh-483) [inline pkg/drivers/http/client.go]: same as above

File: pkg/guid/guid_pool.go [hunk 7]
```diff
@@ -87,7 +87,7 @@ func (p *guidPool) InitPool() error {
 
 	for _, pod := range pods.Items {
 		glog.V(3).Infof("InitPool(): checking pod for network annotations %v", pod)
-
+		pod := pod // pin!
```
Comment (Mmduh-483) [inline pkg/guid/guid_pool.go]: `// pin!` is it related to the lint? and why re-define the var pod again?

File: pkg/sm/plugins/noop/noop.go [hunk 8]
```diff
@@ -1,9 +1,11 @@
 package main
 
 import (
-	"github.com/Mellanox/ib-kubernetes/pkg/sm/plugins"
-	"github.com/golang/glog"
 	"net"
+
+	"github.com/golang/glog"
+
+	"github.com/Mellanox/ib-kubernetes/pkg/sm/plugins"
```
Comment (Mmduh-483) [inline pkg/sm/plugins/noop/noop.go]: lines 6 and 8 should be swapped

File: pkg/utils/synchronized_map.go [hunk 9]
```diff
@@ -6,7 +6,7 @@ import (
 
 // SynchronizedMap a thread safe string to anything map
 type SynchronizedMap struct {
-	Items map[string]interface{}
+	Items        map[string]interface{}
```
Comment (Mmduh-483) [inline pkg/utils/synchronized_map.go]: Is it related to the lint?

File: pkg/watcher/resource-event-handler/pod.go [hunk 10]
```diff
@@ -1,4 +1,4 @@
-package resouce_event_handler
+package resource_event_handler
```
Comment (Mmduh-483) [inline pkg/watcher/resource-event-handler/pod.go]: IMHO: It should be in another commit

File: pkg/watcher/watcher_test.go [hunk 11]
```diff
@@ -1,13 +1,15 @@
 package watcher
 
 import (
-	resEventHandler "github.com/Mellanox/ib-kubernetes/pkg/watcher/resouce-event-handler"
 	"time"
 
-	k8sClientMock "github.com/Mellanox/ib-kubernetes/pkg/k8s-client/mocks"
-	"github.com/Mellanox/ib-kubernetes/pkg/watcher/resouce-event-handler/mocks"
```
Comment (Mmduh-483) [inline pkg/watcher/watcher_test.go]: Why move them to other section?

File: pkg/watcher/watcher_test.go [hunk 12]
```diff
@@ -1,13 +1,15 @@
...
+	resEventHandler "github.com/Mellanox/ib-kubernetes/pkg/watcher/resource-event-handler"
+
 	"github.com/stretchr/testify/mock"
```
Comment (Mmduh-483) [inline pkg/watcher/watcher_test.go]: Should move to 3rd party section.

File: Makefile (hunk 3 above)
Comment (moshe010) [reply Makefile]: because golangci support it build it see golangci.yml in the commit

File: cmd/ib-kubernetes/main.go (hunk 4 above)
Comment (moshe010) [reply cmd/ib-kubernetes/main.go]: no, we the first define standard lib, thirdpart and then local libs as define in goimports

File: pkg/daemon/daemon.go (hunk 1 above)
Comment (moshe010) [reply pkg/daemon/daemon.go]: It is in the third party section

File: pkg/drivers/http/client.go (hunk 5 above)
Comment (moshe010) [reply pkg/drivers/http/client.go]: it relate to mspell lint

File: pkg/drivers/http/client.go (hunk 6 above)
Comment (moshe010) [reply pkg/drivers/http/client.go]: it related to bodyclose lint

File: pkg/utils/synchronized_map.go (hunk 9 above)
Comment (moshe010) [reply pkg/utils/synchronized_map.go]: yes

File: pkg/watcher/resource-event-handler/pod.go (hunk 10 above)
Comment (moshe010) [reply pkg/watcher/resource-event-handler/pod.go]: again it related to the mspell lint

**Comment by moshe010** [comment]
@adrianchiris, the configuration is not finalize I just pushed what was easy to fix in the repository.
We will add other lint in the near future.
